WORKSPACE

```diff
@@ -0,0 +1,51 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "rules_foreign_cc",
+    sha256 = "6041f1374ff32ba711564374ad8e007aef77f71561a7ce784123b9b4b88614fc",
+    strip_prefix = "rules_foreign_cc-0.8.0",
+    url = "https://github.com/bazelbuild/rules_foreign_cc/archive/0.8.0.tar.gz",
+)
+
+load("@rules_foreign_cc//foreign_cc:repositories.bzl", "rules_foreign_cc_dependencies")
+
+rules_foreign_cc_dependencies()
+
+http_archive(
```

Comment:
I see. It seems like bazel is not designed to build libraries that can be installed on a system.

Would it be correct to say that the `WORKSPACE` file here is only useful for running KJ's own CI? Any dependent projects would not use this, and would instead only use the inner `BUILD` files?

Reply:
That's correct @kentonv. Most projects by convention expose a `deps.bzl` file which contains a macro for loading the required dependencies of the project. They can be override by the root workspace if required. WORKSPACE files for external repositories are not evaluated. 

This should become simpler with the introduction of bzlmod, though it's still in beta.

Reply:
I was slightly wrong. This WORKSPACE file won't be used in deps:

```
Bazel only reads dependencies listed in your WORKSPACE file. If your project (A) depends on another project (B) which lists a dependency on a third project (C) in its WORKSPACE file, you'll have to add both B and C to your project's WORKSPACE file. This requirement can balloon the WORKSPACE file size, but limits the chances of having one library include C at version 1.0 and another include C at 2.0.
```
https://bazel.build/docs/external

if we don't publish deps.bzl like @uhthomas suggested, then I believe all users of capnp will need to provide repositories with names like `@ssl` or `@zlib` themselves.

Reply:
I think it would be wrong to not provide some sort of macro to declare dependencies. We also rely on skylib, which need to be pulled in from somewhere. Documentation feels like a bad solution.

I see two options really:

1. Declare _some_ dependency of OpenSSL and zlib, and allow users to override that version if they want. The convention in Bazel before bzlmod is to declare those dependencies before calling the macro. Repositories can be declared multiple times, but the first declaration wins essentially. The `maybe` macro from Bazel tools ensures that is the case. The advantage of this is causal users won't need worry about finding their own sources for OpenSSL and zlib, lowering the barrier for usage.

2. As suggested previously, require those dependencies be given explicitly to a toolchain. This means that OpenSSL and zlib cannot be accidentally missed and their repository names and targets won't have to be hard-coded. Also allows for different versions to be used depending on toolchain resolution.

Reply:
Can we keep it simple for a while (as here) and worry about making it easier to use once there's a real request?